test/connectivity/connectivity.spec.js

```diff
@@ -0,0 +1,305 @@
+const { expect } = require('chai');
+const Enzyme = require('enzyme');
+const Adapter = require('enzyme-adapter-react-16');
+const jsdomGlobal = require('jsdom-global');
+const m = require('module');
+const React = require('react');
+const debug = require('debug')('connectivity-test');
+const { promisify } = require('util');
+const sinon = require('sinon');
+
+Enzyme.configure({ adapter: new Adapter() });
+const { mount } = Enzyme;
+
+// Stub electron module for tests
+const originalLoader = m._load;
+const stubs = {
+  electron: {
+    ipcMain: {
+      call: (methodName) => {
+        debug('electron.ipcMain.call main called!', methodName);
+      },
+      respondTo: (methodName) => {
+        debug('electron.ipcMain.respondTo main called!', methodName);
+      },
+      on: (methodName) => {
+        debug('electron.ipcMain.on main called!', methodName);
+      }
+    },
+    remote: {
+      app: {
+        getName: () => 'Compass Connectivity Integration Test Suite',
+        getPath: () => ''
+      },
+      dialog: {
+        remote: {},
+        BrowserWindow: {}
+      }
+    },
+    shell: {}
+  }
+};
+m._load = function hookedLoader(request, parent, isMain) {
+  const stub = stubs[request];
+  return stub || originalLoader(request, parent, isMain);
+};
+
+// Mock dom/window for tests.
+// Has to come before @mongodb-js/compass-connect require.
+jsdomGlobal();
+
+const AppRegistry = require('hadron-app-registry');
+const Connection = require('mongodb-connection-model');
+const CompassConnectPlugin = require('@mongodb-js/compass-connect');
+const CompassConnectComponent = CompassConnectPlugin.default;
+const activateCompassConnect = CompassConnectPlugin.activate;
+
+const deactivateCompassConnect = CompassConnectPlugin.deactivate;
+
+const {
+  dockerComposeDown,
+  dockerComposeUp
+} = require('./docker-instance-manager');
+
+// Hide react warnings.
+// eslint-disable-next-line no-console
+const originalWarn = console.warn.bind(console.warn);
+// eslint-disable-next-line no-console
+console.warn = (msg) => (
+  !msg.toString().includes('componentWillReceiveProps')
+  && !msg.toString().includes('componentWillUpdate')
+  && originalWarn(msg)
+);
+
+const delay = promisify(setTimeout);
+
+const connectionsToTest = [{
+  title: 'local community standalone',
+  dockerComposeFilePath: 'community/docker-compose.yaml',
+  model: {
+    hostname: 'localhost',
+    port: 27020,
+  },
+  expectedInstanceDetails: {
+    _id: 'localhost:27020',
+    authenticatedUsers: [],
+    build: {
+      raw: {
+        version: /4.4.[1-9]+$/
+      }
+    },
+    client: {
+      isWritable: true,
+      isMongos: false,
+    },
+    dataLake: { isDataLake: false, version: null },
+    genuineMongoDB: { isGenuine: true, dbType: 'mongodb' },
+    host: {
+      os: 'Ubuntu',
```

Comment:
doesn't this break locally?

Reply:
That's the host of the docker image, which should be ubuntu. Tests pass locally on my mac 👍

Reply:
Thinking about that though - hopefully we can get one of our tests running on a host machine that isn't ubuntu for test sake.